include/nanobind/nb_traits.h

```diff
@@ -114,6 +114,9 @@ template <template <typename> typename Op, typename Arg>
 struct detector<std::void_t<Op<Arg>>, Op, Arg>
     : std::true_type { };
 
+// Specialized in stl/optional.h and stl/variant.h
```

Comment:
Improve comment for posterity:
```
/* This template is used for docstring generation and specialized in
   ``stl/{variant,optional.h}`` to strip away std::optional and
   ``std::variant<std::monostate>`` in top-level argument types and
   avoid redundancy when combined with nb::arg(...).none(). */
```

Also, note: space between `template` and `<typename` matching other lines.